common-jvm/src/main/java/org/ergoplatform/transactions/ColdWalletUtils.kt

```diff
@@ -238,7 +238,7 @@ fun buildTransactionInfoFromReduced(
         val inputInfo = InputInfo()
         inputInfo.boxId = boxid
         inputBoxes.get(boxid)?.let {
-            inputInfo.address = Address.fromErgoTree(it.ergoTree(), ergoNetworkType).toString()
+            inputInfo.address = Address.fromErgoTree(it.ergoTree(), getErgoNetworkType()).toString()
```

Comment:
It is implicitely done: boxId variable holds the id from unsignedTx.inputs and inputBoxes.get() retrieves it only when the id is the same. If the information is missing in inputBoxes, only the box id is shown (more can't be done). If inputBoxes holds box IDs that aren't actually used, they are ignored. So it is guarantueed that the information shown to the user is valid.

Reply:
I think this is not enough to rely on user attentiveness alone. I think it is better to reject ill-formed transactions is then inputBoxes and unsigned boxes don't match exactly.

Reply:
Okay, in that case we should update the EIP that the inputs list is not optional? Or do we want to allow inputs list to be optional and allow showing no input information allow, but restrict incomplete lists?
In any case, this should be moved to an own issue for the changes since the file changes in this PR are only done because the file was moved and is not used on iOS. Furthermore, hardening the validation needs UI changes in Android which is the wrong place here.

Reply:
It can be a separate issue, but it is better to be solved before release.

Reply:
I've added issue #58 please approve this PR when it is okay for now.